pkg/format/format.go

```diff
@@ -21,36 +20,133 @@ const (
 )
 
 var (
-	labelVar  string
-	fsTypeVar string
-	drives    map[string]bool
-	driveKeys []string
+	labelVar   string
+	fsTypeVar  string
+	forceVar   bool
+	verboseVar bool
+	drives     map[string]bool
+	driveKeys  []string
 )
 
+func hasPartitions(d string) bool {
+	err := exec.Command("sfdisk", "-d", d).Run()
+	return err == nil
+}
+
+func isEmptyDevice(d string) (bool, error) {
+	// default result
+	isEmpty := false
+
+	if verboseVar {
+		log.Printf("Checking if %s is empty", d)
+	}
+
+	out, err := exec.Command("blkid", d).Output()
+	if err == nil {
+		log.Printf("%s has content. blkid returned: %s", d, out)
+		// there is content, so exit early
+		return false, nil
+	}
+
+	if exiterr, ok := err.(*exec.ExitError); ok {
+		if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+			// blkid exitcode 2 (from the non-busybox version) signifies the block device has no detectable content signatures
+			if status.ExitStatus() == 2 {
+				if verboseVar {
+					log.Printf("blkid did not find any existing content on %s.", d)
+				}
+				// no content detected, but continue through all checks
+				isEmpty = true
+			} else {
+				return isEmpty, fmt.Errorf("Could not determine if %s was empty. blkid %s returned: %s (exitcode %d)", d, d, out, status.ExitStatus())
+			}
+		}
+	}
+
+	if hasPartitions(d) {
+		log.Printf("Partition table found on device %s. Skipping.", d)
+		return false, nil
+	}
+
+	// return final result
+	return isEmpty, nil
+}
+
 func autoformat(label, fsType string) error {
 	var first string
 	for _, d := range driveKeys {
-		err := exec.Command("sfdisk", "-d", d).Run()
+		if verboseVar {
+			log.Printf("Considering auto format for device %s", d)
+		}
+		// break the loop with the first empty device we find
+		isEmpty, err := isEmptyDevice(d)
```

Comment:
Since the code below needs changing, might as well refactor this as well to
```
if isEmpty, err := isEmptyDevice(d); err != nil {
  return err
} else {
  if isEmpty == true {
    first = d
    break
}
```
This limits the scope of `isEmpty`

Reply:
Doing so now. Good point.